generator/internal/discovery_file.cc

```diff
@@ -93,14 +96,19 @@ package $package_name$;
   return {};
 }
 
-Status DiscoveryFile::WriteFile(std::string const& product_name) const {
+Status DiscoveryFile::WriteFile(
+    std::string const& product_name,
+    std::map<std::string, DiscoveryTypeVertex> const* types) const {
+  if (types == nullptr) {
+    return internal::InternalError("types is nullptr");
+  }
```

Comment:
comment: It is a little weird to have `types` be a pointer, when `nullptr` is invalid... but I guess you need to put the check somewhere and `WriteFile(...)` is most convenient.